api/auth/auth-api.ts

```diff
@@ -0,0 +1,5 @@
+export interface AuthApi {
```

Comment:
```suggestion
export interface AccessTokenProvider {
```
implementation probably change to `StaticAccessTokenProvider`

and IMHO SmartlingAuthApi should not extend ~AuthApi~ AccessTokenProvider, 
use delegation instead
```
export class DynamicAccessTokenProvider implements AccessTokenProvider {
    constructor(
        private authApi: SmartlingAuthApi
    ) {}
    
  async getAccessToken(): Promise<string> {
        return authApi.getAccessToken();
    }
}
```

Reply:
Agree with naming. will rework.

> and IMHO SmartlingAuthApi should not extend AuthApi AccessTokenProvider,
use delegation instead

it's another story. not part of this task.